### PR TITLE
refactor(operator): improve extra args injection readability

### PR DESCRIPTION
**Pull request by bwplotka**
Title should be self-explanatory - small change I found useful, but don't want to bloat other PRs with this.

### CODE REVIEW

File: pkg/operator/collection.go [hunk 1]
```diff
@@ -267,23 +267,8 @@ func (r *collectionReconciler) ensureCollectorDaemonSet(ctx context.Context, spe
 	if len(spec.Compression) > 0 && spec.Compression != monitoringv1.CompressionNone {
 		flags = append(flags, fmt.Sprintf("--export.compression=%s", spec.Compression))
 	}
+	setContainerExtraArgs(ds.Spec.Template.Spec.Containers, "prometheus", strings.Join(flags, " "))
```
Comment (TheSpiritXIII) [inline pkg/operator/collection.go:270]: nit: use `CollectorPrometheusContainerName`

File: pkg/operator/operator_config.go [hunk 2]
```diff
@@ -415,23 +437,7 @@ func (r *operatorConfigReconciler) ensureAlertmanagerStatefulSet(ctx context.Con
...
-		for _, ev := range c.Env {
-			if ev.Name != "EXTRA_ARGS" {
-				repl = append(repl, ev)
-			}
-		}
-		repl = append(repl, corev1.EnvVar{Name: "EXTRA_ARGS", Value: strings.Join(flags, " ")})
-
-		sset.Spec.Template.Spec.Containers[i].Env = repl
-	}
+	setContainerExtraArgs(sset.Spec.Template.Spec.Containers, "alertmanager", strings.Join(flags, " "))
```
Comment (TheSpiritXIII) [inline pkg/operator/operator_config.go:440]: nit: use `AlertmanagerContainerName`.

File: pkg/operator/operator_config.go [hunk 3]
```diff
@@ -477,23 +483,7 @@ func (r *operatorConfigReconciler) ensureRuleEvaluatorDeployment(ctx context.Con
...
-		for _, ev := range c.Env {
-			if ev.Name != "EXTRA_ARGS" {
-				repl = append(repl, ev)
-			}
-		}
-		repl = append(repl, corev1.EnvVar{Name: "EXTRA_ARGS", Value: strings.Join(flags, " ")})
-
-		deploy.Spec.Template.Spec.Containers[i].Env = repl
-	}
+	setContainerExtraArgs(deploy.Spec.Template.Spec.Containers, "evaluator", strings.Join(flags, " "))
```
Comment (TheSpiritXIII) [inline pkg/operator/operator_config.go:486]: nit: use `RuleEvaluatorContainerName`.

**Review by TheSpiritXIII — APPROVED** [review]
Great cleanup! Just a minor nit. :)

**Review by pintohutch — APPROVED** [review]

**Pull request merged**